editor/header/saved-state/index.js

```diff
@@ -2,51 +2,73 @@
  * External dependencies
  */
 import { connect } from 'react-redux';
-import classNames from 'classnames';
+import classnames from 'classnames';
 
 /**
  * WordPress dependencies
  */
-import { Dashicon } from 'components';
+import { __ } from 'i18n';
+import { Dashicon, Button } from 'components';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
-import { isEditedPostNew, isEditedPostDirty } from '../../selectors';
+import { savePost } from '../../actions';
+import {
+	isEditedPostNew,
+	isEditedPostDirty,
+	isSavingPost,
+	getCurrentPost,
+	getPostEdits,
+	getBlocks,
+	getEditedPostStatus,
+} from '../../selectors';
 
-function SavedState( { isNew, isDirty } ) {
-	const classes = classNames( 'editor-saved-state', {
-		'is-new': isNew,
-		'is-dirty': isDirty,
-	} );
+function SavedState( { isNew, isDirty, isSaving, edits, blocks, post, status, onSave } ) {
+	const className = 'editor-saved-state';
 
-	let icon, text;
-	if ( isNew && isDirty ) {
-		icon = 'warning';
-		text = wp.i18n.__( 'New post with changes' );
-	} else if ( isNew ) {
-		icon = 'edit';
-		text = wp.i18n.__( 'New post' );
-	} else if ( isDirty ) {
-		icon = 'warning';
-		text = wp.i18n.__( 'Unsaved changes' );
-	} else {
-		icon = 'saved';
-		text = wp.i18n.__( 'Saved' );
+	if ( isSaving ) {
+		return (
+			<span className={ className }>
+				{ __( 'Saving' ) }
+			</span>
+		);
 	}
+	if ( ! isNew && ! isDirty ) {
+		return (
+			<span className={ className }>
+				<Dashicon icon="saved" />
+				{ wp.i18n.__( 'Saved' ) }
```

Comment:
Now that we're importing `__`, should we be consistent with how we're calling it throughout the file?